engine/classes/ElggPlugin.php

```diff
@@ -539,17 +544,12 @@ public function isValid() {
 	 * @return bool
 	 */
 	public function isActive() {
-		if (!$this->guid) {
-			return false;
-		}
-
-		$site = elgg_get_site_entity();
-
-		if (!($site instanceof \ElggSite)) {
-			return false;
+		if (isset($this->activated)) {
+			return $this->activated;
 		}
 
-		return check_entity_relationship($this->guid, 'active_plugin', $site->guid) instanceof ElggRelationship;
+		$this->activated = elgg_is_active_plugin($this->getID());
```

Comment:
I don't like this. I'd prefer not to use global APIs within object instances. It should be the other way around. `elgg_is_active_plugin()` should load plugin entity and check `ElggPlugin::isActive`

Reply:
tried it... can't figure out a good way... you can give it a try if you think there is a way

Reply:
only the service nows activated vs inactivated plugins... letting the plugin figure it out will cost unnecessary queries

Reply:
Not a priority at the moment. Could you add a todo somewhere?

Reply:
todo is here https://github.com/Elgg/Elgg/issues/11827